File: code_samples/scala/data_structures_hash_table.scala
/**
 * Data Structures: Hash Table
 * AI/ML Training Sample
 */

class HashTable {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object HashTableExample {
  def main(args: Array[String]): Unit = {
    val instance = new HashTable()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/lua/data_structures_hash_table.lua
-- Data Structures: Hash Table
-- AI/ML Training Sample

HashTable = {}
HashTable.__ index = HashTable

function HashTable:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function HashTable:process(input)
    self.data = input
end

function HashTable:getData()
    return self.data
end

function HashTable:validate()
    return #self.data > 0
end

-- Example usage
local instance = HashTable:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
